### DISCUSSION TITLE
spell checker in markdown is very messy

### DISCUSSION THREAD

**Discussion by regevbr** [requarks/wiki#3546] [Error / Bug Report]
**Describe the bug**
When enabling the spell checker in markdown, the errors only shown in the preview panel.
When fixing a spelling mistake in the editor, the focus is lost and you need to enable the spell checker again. Doing this over and over again is very annoying and time-consuming.
In addition, the spell checker doesn't work for rendered lists or any other special UX elements.

**To Reproduce**
1. Take a long markdown page and add a lot of spell errors all over it
2. Try to fix those errors as fast as you can
3. Stop when you are frustrated

**Expected behavior**
Hopefully, we will be able to perform the spell check on the editor itself and not the preview so we can let the spell checker perform the fixes for us inline. If that is not possible, at least not lose focus on every change to the editor...
I think the solution I suggested at https://github.com/Requarks/wiki/issues/2094#issue-642943771 will be best

**Host Info (please complete the following information):**
 - OS: Docker
 - Wiki.js version: 2.5.117
 - Database engine: postgres 11.6

**Comment by regevbr** [comment]
@NGPixel what do you think? I don't mind working on a PR to get that rolling fast

**Comment by regevbr** [comment]
I found out that we can set the following properties when initiating CodeMirror:
```json
      spellcheck: true,
      autocorrect: true,
      autocapitalize: true,
```
And then the browser spellchecker can kick in inside of the code editor!
I did find out though that it doesn't play nicely with Grammarly, so I think that enabling a pure text box vide mode is still the better solution.
Also, dynamically changing the code mirror attributes will be harder (if even possible)

**Reply by T3P3** [reply]
@regevbr did your pull request for this ever progress?

**Reply by regevbr** [reply]
Sadly i did not

**Reply by T3P3** [reply]
Is this a complex change? I may be missing some interaction or other that means is is not easy to implement but in the face of it, it appears that your solution fixed the problem.

**Reply by regevbr** [reply]
It just might but i never got around to it

**Reply by T3P3** [reply]
ok, thanks, just wanted to see if I was missing something important. What's the protocol with wiki.js as far as raising issues? - raise a new issue to point that there is a solution shown here? The discussion was raised from an issue initially.

**Reply by regevbr** [reply]
You can create a pr but the maintainer is very busy. I have a couple of open prs for a year now without any progress so don't keep your hopes up

**Comment by D1reW0lf** [comment]
Just to add a "me too" on this.

Wiki.js version:  2.5.201
Node.js: 14.16.0
Database engine: postgres 11.11
OS: Docker container (Linux)

**Comment by T3P3** [comment]
I can also confirm that changing the codemirror setting to` spellcheck="true"` fixed this issue in the latest chrome.

![image](https://user-images.githubusercontent.com/1642947/137904052-fdb2277b-6743-4270-b5f2-780cee80bcc2.png)

This also allows for the normal spellcheck interaction with the bowser (right check, select suitable replacement) to work as expected.

Is there a reason this can't be used for codemirror, if not as a default then at least as a selectable option from the admin panel?

**Comment by deggle** [comment]
Is there any reason this hasn't been progressed? Not having any spell check is such a fundamental issue and it looks like it may be fixable with a single line of code!?!?
